Include/internal/pycore_frame.h

```diff
@@ -148,17 +148,26 @@ _PyFrame_NumSlotsForCodeObject(PyCodeObject *code)
 
 static inline void _PyFrame_Copy(_PyInterpreterFrame *src, _PyInterpreterFrame *dest)
 {
-    *dest = *src;
+    dest->f_executable = PyStackRef_MakeHeapSafe(src->f_executable);
+    // Don't leave a dangling pointer to the old frame when creating generators
+    // and coroutines:
+    dest->previous = NULL;
+    dest->f_funcobj = PyStackRef_MakeHeapSafe(src->f_funcobj);
+    dest->f_globals = src->f_globals;
+    dest->f_builtins = src->f_builtins;
+    dest->f_locals = src->f_locals;
+    dest->frame_obj = src->frame_obj;
+    dest->instr_ptr = src->instr_ptr;
+#ifdef Py_GIL_DISABLED
+    dest->tlbc_index = src->tlbc_index;
+#endif
     assert(src->stackpointer != NULL);
     int stacktop = (int)(src->stackpointer - src->localsplus);
-    assert(stacktop >= _PyFrame_GetCode(src)->co_nlocalsplus);
+    assert(stacktop >= 0);
     dest->stackpointer = dest->localsplus + stacktop;
-    for (int i = 1; i < stacktop; i++) {
-        dest->localsplus[i] = src->localsplus[i];
+    for (int i = 0; i < stacktop; i++) {
+        dest->localsplus[i] = PyStackRef_MakeHeapSafe(src->localsplus[i]);
     }
```

Comment:
What's the story behind the changed assert and the different starting index?

Reply:
Previously we used memcpy to copy the bulk of the frame. Since the struct ends with `localsplus[1];` the memcpy copied the first local. Now we copy the other fields explicitly, so we need to copy the `localsplus[0]` explicitly as well.